src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -96,7 +97,13 @@ public void save(AddressBook addressBook) throws StorageOperationException {
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
-        try (final Writer fileWriter =
+        
+    	File f = new File (DEFAULT_STORAGE_FILEPATH);
+    	if (!(f.exists())) {
+    		throw new StorageOperationException("Error finding file: " + path);
```

Comment:
i hope you remember the discussion during the tutorial.
you have got the idea right; however, the implementation is a erroneous